packages/mds-metrics/api.ts

```diff
@@ -54,7 +57,42 @@ function api(app: express.Express): express.Express {
 
   app.get(
     pathsFor('/all'),
-    checkAccess(scopes => scopes.includes('admin:all')),
+    async (req, res, next) => {
+      if (res.locals.scopes.includes('metrics:read:provider')) {
+        if (!res.locals.claims) {
+          return res.status(400).send({
+            result: 'No claims provided'
+          })
+        }
+        const { provider_id } = res.locals.claims
+
+        if (!isUUID(provider_id)) {
+          await log.warn(req.originalUrl, 'invalid provider_id is not a UUID', provider_id)
+          return res.status(400).send({
+            result: `invalid provider_id ${provider_id} is not a UUID`
+          })
+        }
+
+        if (!isProviderId(provider_id)) {
+          return res.status(400).send({
+            result: `invalid provider_id ${provider_id} is not a known provider`
+          })
+        }
+
+        // stash provider_id
+        res.locals.provider_id = provider_id
+        // res.locals.provider_ids must contain provider_id from claim
+        // other queried providers are ignored
+        res.locals.provider_ids = [provider_id]
+      } else if (res.locals.scopes.includes('metrics:read')) {
+        const provider_ids = normalizeToArray<UUID>(req.query.provider_id)
+        res.locals.provider_ids = provider_ids
+      } else {
+        return res.status(401).send('Unauthorized')
+      }
+      next()
+    },
+    checkAccess(scopes => scopes.includes('metrics:read') || scopes.includes('metrics:read:provider')),
```

Comment:
This should go before the above function. Make sure one of the two scopes is present first.